### PR TITLE
migration/changes for nextcloud 23/24, bump UserExternal v3.0.0

### LINKED ISSUE #186: Nextcloud v23 support
External User Backends (user_external) supports nextcloud 21 and 22, but not yet 23. 

Is support planned? 

Has anyone tried to enable this app on nc23 despite the lack of official support?

EDIT: [The official nextcloud v23 documentation points to this unsupported app](https://docs.nextcloud.com/server/23/admin_manual/configuration_user/user_auth_ftp_smb_imap.html)

### PR DESCRIPTION
**Pull request by MarBie77**
Signed-off-by: Biermair Martin <martin.biermair@wimbergerhaus.at>

Fixes #186 

Changes proposed in this pull request:
 - increased supported nextcloud major version to 24
 - migrating app.php to AppInfo/Application class
 - increased to version 3.0.0, because breaking namespace changes
 - adopting README and tests

### CODE REVIEW

**Comment by VVD** [comment]
How to migrate existing external users? Didn't find the description in the README.md.
Thanks!

**Comment by MarBie77** [comment]
There is no need to migrate the users. But you have to change your config to the new namespace (class names), see README.md.

**Comment by VVD** [comment]
> There is no need to migrate the users.

Ok, thanks.
> But you have to change your config to the new namespace (class names), see README.md.

Yes, already read this.

**Comment by violoncelloCH** [comment]
thanks for your PR @MarBie77 !
@ChristophWurst could you maybe take a quick look and tell me if this seems good to you? (I'm not really up to date on the latest Nextcloud best practices regarding app development)
I'd then create a new release this week-end (possibly also for Nextcloud 24)

File: README.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 External user authentication
 ============================
+
+**⚠⚠ Warning:** As of Version 3.0 this app uses namespace \OCA\UserExternal now. You MUST change your config to adopt to this change. ⚠⚠
+
+## Usage
+
+Go to Apps-Folder of your Nextcloud Installation
```
Comment (violoncelloCH) [inline README.md:8]: @MarBie77 what's the meaning of this phrase? 
the configuration has to be done in `config/config.php`...

File: appinfo/info.xml [hunk 2]
```diff
@@ -32,6 +32,6 @@ Read the [documentation](https://github.com/nextcloud/user_external#readme) to l
 	<bugs>https://github.com/nextcloud/user_external/issues</bugs>
 	<repository type="git">https://github.com/nextcloud/user_external.git</repository>
 	<dependencies>
-		<nextcloud min-version="21" max-version="22" />
+		<nextcloud min-version="21" max-version="23" />
```
Comment (violoncelloCH) [inline appinfo/info.xml:35]: let's drop support for Nextcloud 21 (it's EOL) and add support for 24 instead

File: README.md (hunk 1 above)
Comment (MarBie77) [reply README.md]: Sorry, removed the additional text from a former version!

**Comment by skjnldsv** [comment]
@MarBie77 didn't see that PR, coul dyou trake the workflows/composer from https://github.com/nextcloud/user_external/pull/192 and add them here please?
You can also remove drone

**Review by CarlSchwan — APPROVED** [review]

**Comment by violoncelloCH** [comment]
I just tried to activate user_external from this branch, but get 
```
{"reqId":"8n52aJhC41keGQWqkj5z","level":3,"time":"2022-04-14T13:40:03+00:00","remoteAddr":"192.168.21.3","user":"admin","app":"PHP","method":"POST","url":"/index.php/settings/apps/enable","message":"Cannot declare class OCA\\UserExternal\\AppInfo\\Application, because the name is already in use at /var/www/html/apps/user_external/lib/AppInfo/Application.php#14","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:99.0) Gecko/20100101 Firefox/99.0","version":"24.0.0.8"}
```
when clicking on activate in the Apps...
not sure if this is just an issue with my dev setup though :thinking: ?

**Comment by violoncelloCH** [comment]
okay I think I found it and pushed a fix for it... I've also increased the version number to 3.0.0

**Review by violoncelloCH — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**